add cnspec scan and cnspec shell examples (#33)
mondoohq/docs: docs/cnspec/cli/cnspec_scan.md

title: cnspec scan
---

Run a security scan on an asset based on one or more Mondoo policies.

To learn more, read [Get Started with cnspec](/cnspec/).

### Synopsis

> This command triggers a new policy-based scan on an asset. By default, cnspec scans the local system with the default [policies](/cnspec/cnspec-policies/) built specifically for the platform:

Reviewer: I'm not sure if this is describing the process correctly. By default (when logged in) you're getting policies enabled in your space

Author: Thanks. I pushed a change. Is that better?

```bash
cnspec scan local
```

> You can also specify a local policy and run it without storing results in the server:

Reviewer: server or platform?

Author: Good catch. That's old!

> You can also specify a local policy and run it without storing results in the server:

Author: ```suggestion
You can also specify a local policy and run it without storing results in Mondoo Platform:
```

```

In addition, cnspec can scan assets remotely using SSH. By default, cnspec uses the operating system's SSH agent and SSH config to retrieve the credentials:

```bash
cnspec scan ssh ec2-user@52.51.185.215
```

```bash
cnspec scan ssh ec2-user@52.51.185.215:2222
```

### Examples: cloud

#### Scan AWS

```bash
cnspec scan aws --region us-east-1
```

To learn more, read [Assess AWS Security with cnspec](/cnspec/cnspec-aws/).

#### Scan Azure

```bash
cnspec scan azure --subscription SUBSCRIPTION_ID --group GROUP_NAME
```

To learn more, read [Assess Azure Security with cnspec](/cnspec/cnspec-azure/).

#### Scan Google Cloud (GCP)

```bash
cnspec scan gcp project PROJECT_ID
```

To learn more, read [Assess Google Cloud Security with cnspec](/cnspec/cnspec-gcp/).

#### Scan Kubernetes

```bash
cnspec scan k8s
```

```bash
cnspec scan k8s MANIFEST_FILE
```

To learn more, read [Assess Kubernetes Security with cnspec](/cnspec/cnspec-k8s/).

#### Scan Oracle Cloud Infrastructure (OCI)

```bash
cnspec scan oci
```

To learn more, read [Assess Oracle Cloud Infrastructure (OCI) Security with cnspec](/cnspec/cnspec-oci/).

### Examples: SaaS

> #### Scan GitHub

Reviewer: GitLab too in this section?

Author: GitLab docs are coming.

```bash
export GITHUB_TOKEN=YOUR_PERSONAL_ACCESS_TOKEN
cnspec scan github repo ORG/REPO
```

To learn more, read [Assess GitHub Security with cnspec](/cnspec/saas/github/).

#### Scan Google Workspace

```bash
export GOOGLEWORKSPACE_CLOUD_KEYFILE_JSON=/home/user/my-project-6646123456789.json
cnspec scan google-workspace --customer-id 5amp13iD --impersonated-user-email admin@domain.com
```

To learn more, read [Assess Google Workspace Security with cnspec](/cnspec/saas/google_workspace/).

#### Scan Microsoft 365 (MS 365)

```bash
cnspec scan ms365 --certificate-path certificate.combo.pem --tenant-id YOUR_TENANT_ID --client-id YOUR_CLIENT_ID
```

To learn more, read [Assess Microsoft 365 Security with cnspec](/cnspec/saas/ms365/).

#### Scan Okta

```bash
cnspec scan okta --organization your_org.okta.com --token API_TOKEN
```

To learn more, read [Assess Okta Security with cnspec](/cnspec/saas/okta/).

#### Scan Slack

```bash

> cnspec shell slack --token API_TOKEN

Reviewer: ```suggestion
cnspec scan slack --token API_TOKEN
```

```

To learn more, read [Assess Slack Security with cnspec](/cnspec/saas/slack/).

### Examples: supply chain and containers

cnspec supports local containers and images as well as images in Docker registries.

> #### Scan Docker

Reviewer: It might be helpful here to say Docker containers since we're not scanning the actual app runtime

```bash
cnspec scan docker container b62b276baab6
```

```bash
cnspec scan docker image ubuntu:latest
```

#### Scan Harbor

```bash
cnspec scan container registry harbor.lunalectric.com
```

#### Scan ECR

```bash
cnspec scan container registry 123456789.dkr.ecr.us-east-1.amazonaws.com/repository
```

#### Scan GCR

```bash
cnspec scan gcp gcr PROJECT_ID
```

#### Scan Vagrant

```bash
cnspec scan vagrant HOST
```

> #### Scan an inventory file

Reviewer: It might be worth mentioning here what an inventory file actually is and linking to those docs

```bash
cnspec scan --inventory-file inventory.yml
```

> #### Scan Ansible

Reviewer: ```suggestion
#### Scan Ansible inventory
```
